src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java

```diff
@@ -3396,4 +3398,40 @@ void subscribeExclusiveTopicAndPattern(VertxTestContext context) throws Throwabl
         context.completeNow();
         assertTrue(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS));
     }
+
+    @Test
+    void readyTest(VertxTestContext context) throws InterruptedException {
+        int iterations = 5;
+        for (int i = 1; i <= iterations; i++) {
+            getRequest("/ready")
+                    .send(ar -> {
+                        context.verify(() -> {
+                            assertTrue(ar.succeeded());
+                            assertEquals(HttpResponseStatus.OK.code(), ar.result().statusCode());
+                        });
+                    });
+            Thread.sleep(1000);
+            if (i == iterations) {
+                context.completeNow();
+            }
```

Comment:
because it's all async it could happen that if the last call to the `/ready` endpoint takes more than 2 seconds, you are completing the context before the actual outcome from the GET request.
it means that you have to do the check inside the handler.